<a href="https://colab.research.google.com/github/aryaabinash/Python-Automation/blob/master/DataPreprocessing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Part 1 - Sanity check - duplicates data


In [2]:
import os
import numpy as np
import pandas as pd
from IPython.display import Image
os.getcwd()


'/content'

In [13]:
students = [('Vikas', 25, 'India'),
            ('Vikas',35,'Syden'),
            ('Anand',16,'Singapore'),
            ('Steve',30,'London'),
            ('Swati',20,'Delhi'),
            ('Anand',16,'Singapore'),
             ('Vikas',35,'Syden')]

df = pd.DataFrame(students, columns = ['Name','Age','City'])
df

Name  Age       City
0  Vikas   25      India
1  Vikas   35      Syden
2  Anand   16  Singapore
3  Steve   30     London
4  Swati   20      Delhi
5  Anand   16  Singapore
6  Vikas   35      Syden

In [23]:
df.duplicated().sum()

2

In [7]:
# Find duplicate rows based on all columns

In [8]:
duplicateRowDF = df[df.duplicated(keep='first')]
print("Duplicate Rows except first occurance on all columns are:")
duplicateRowDF

Duplicate Rows except first occurance on all columns are:


Name  Age       City
5  Anand   16  Singapore
6  Vikas   35      Syden

In [9]:
duplicateRowDF = df[df.duplicated(keep='last')]
print("Duplicate Rows except first occurance on all columns are:")
duplicateRowDF

Duplicate Rows except first occurance on all columns are:


Name  Age       City
1  Vikas   35      Syden
2  Anand   16  Singapore

In [17]:
duplicateRowDF = df[df.duplicated()]
duplicateRowDF

Name  Age       City
5  Anand   16  Singapore
6  Vikas   35      Syden

In [18]:
df[df.duplicated(['Name'],keep='last')]

Name  Age       City
0  Vikas   25      India
1  Vikas   35      Syden
2  Anand   16  Singapore

In [20]:
#Titanic Dataset

In [22]:
titanic = pd.read_csv('titanic_dataset.csv')
titanic.duplicated().sum()

24

In [25]:
titanic.shape

(915, 11)

**How to drop duplicate data**

In [26]:
titanic.drop_duplicates(keep='first',inplace=True)

In [27]:
titanic.shape

(891, 11)

**Missing data treatment**

In [28]:
titanic.isnull().sum()

Survived      0
Pclass        0
Name          0
Sex           0
Age         177
SibSp         0
Parch         0
Ticket        0
Fare          0
Cabin       687
Embarked      2
dtype: int64

In [29]:
titanic.isnull().sum().sum()

866

In [30]:
titanic.isnull().sum()/len(titanic)*100

Survived     0.000000
Pclass       0.000000
Name         0.000000
Sex          0.000000
Age         19.865320
SibSp        0.000000
Parch        0.000000
Ticket       0.000000
Fare         0.000000
Cabin       77.104377
Embarked     0.224467
dtype: float64

Cabin has 77 percent missing value - 77.1% missing data found


In [31]:
titanic = titanic.drop(['Cabin'],axis=1)
# axis = 1 means column wise drop
# axis = 0 means row wise drop
# When there is object no need of writing inplace=True

In [33]:
titanic.isnull().sum()/len(titanic)*100

Survived     0.000000
Pclass       0.000000
Name         0.000000
Sex          0.000000
Age         19.865320
SibSp        0.000000
Parch        0.000000
Ticket       0.000000
Fare         0.000000
Embarked     0.224467
dtype: float64